File: microlearning-adhd-app/frontend/src/services/types/interactionEvent.ts
import { type GroupAssignment, type Subgroup } from './../../utils/groupAssignment'

export type StudyInteractionPayload = Record<string, string | number | boolean | null>

export type StudyInteractionEvent = {
  assignment: GroupAssignment
  subgroup: Subgroup
  page: string
  event_type: string
  occurred_at: string
  payload?: StudyInteractionPayload
}

export type StudyInteractionEventResponse = {
  id: number
  received_at: string
}

File: microlearning-adhd-app/backend/app/schemas/interaction_event.py
from pydantic import BaseModel

from app.timestamps import UtcTimestamp


class InteractionEventRequest(BaseModel):
    assignment: str | None = None
    subgroup: str | None = None
    page: str
    event_type: str
    occurred_at: UtcTimestamp
    payload: dict[str, str | int | float | bool | None] | None = None


class InteractionEventResponse(BaseModel):
    id: int
    received_at: UtcTimestamp
